role_test.go

```diff
@@ -57,6 +58,37 @@ const (
 	deleteRoleResponse  = `{"message":"Role deleted"}`
 )
 
+func TestRoles(t *testing.T) {
+	mockData := strings.Repeat(getRoleResponse+",", 1000) // make 1000 roles.
+	mockData = "[" + mockData[:len(mockData)-1] + "]"     // remove trailing comma; make a json list.
+
+	// This creates 1000 + 1000 + 1 (2001, 3 calls) worth of roles.
+	client := gapiTestToolsFromCalls(t, []mockServerCall{
+		{200, mockData},
+		{200, mockData},
+		{200, "[" + getRoleResponse + "]"},
+	})
+
+	const dashCount = 2001
+
+	roles, err := client.GetRoles()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log(pretty.PrettyFormat(roles))
+
+	if len(roles) != dashCount {
+		t.Errorf("Length of returned roles should be %d", dashCount)
```

Comment:
I believe this should instead be a call to `t.Fatalf`, as otherwise if this one fail then it's possible either of the `if` calls below result in a panic because execution will continue.

Reply:
Good point. I just updated the line with your suggestion